cmd/kubeadm-nspawn/kubeadm-nspawn.go

```diff
@@ -42,6 +44,26 @@ var (
 	baseImage    string
 )
 
+func checkK8sStableRelease(k8srel string) bool {
+	v, err := semver.NewVersion(k8srel)
+	if err != nil {
+		// fallback to default version
+		v, _ = semver.NewVersion(k8sStableVersion)
+	}
+
+	c, err := semver.NewConstraint(">=" + k8sStableVersion)
+	if err != nil {
+		log.Printf("Cannot get constraint for >= %s: %v", k8sStableVersion, err)
+		return false
+	}
+
+	return c.Check(v)
+}
+
+func isK8sUpstream(k8srel string) bool {
```

Comment:
I would call it `isDev()` or use `upstream` as the value to indicate that we mean non-release.

It might be only me but _upstream_ sounds like we're just building the official source. Maybe _local_ (and `isLocal()`) but I'm not sure...

Or a different approach. Could you reuse `checkK8sStableRelease` for that?